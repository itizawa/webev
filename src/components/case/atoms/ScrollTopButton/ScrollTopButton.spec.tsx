import renderer from 'react-test-renderer';
import { ScrollTopButton } from './index';

jest.mock('./hooks', () => ({
  useHooks: () => {
    return { isShowScroll: false, scrollTop: jest.fn() };
  },
}));
test('ScrollTopButton isShowScroll is false', () => {
  const tree = renderer.create(<ScrollTopButton />).toJSON();

  expect(tree).toMatchSnapshot();
});

jest.mock('./hooks', () => ({
  useHooks: () => {
    return { isShowScroll: false, scrollTop: jest.fn() };
  },
}));
test('ScrollTopButton isShowScroll is true', () => {
  const tree = renderer.create(<ScrollTopButton />).toJSON();

  expect(tree).toMatchSnapshot();
});
