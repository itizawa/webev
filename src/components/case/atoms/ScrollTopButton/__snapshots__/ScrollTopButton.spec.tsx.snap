// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollTopButton isShowScroll is false 1`] = `
<button
  className="sc-bdnxRM eJfPil btn btn-light btn-lg show"
  id="scroll-to-top"
  onClick={[MockFunction]}
>
  <svg
    className="align-text-bottom text-secondary"
    fill="currentColor"
    height={18}
    viewBox="0 0 16 16"
    width={18}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15a.5.5 0 0 0 .5-.5V2.707l3.146 3.147a.5.5 0 0 0 .708-.708l-4-4a.5.5 0 0 0-.708 0l-4 4a.5.5 0 1 0 .708.708L7.5 2.707V14.5a.5.5 0 0 0 .5.5z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`ScrollTopButton isShowScroll is true 1`] = `
<button
  className="sc-bdnxRM eJfPil btn btn-light btn-lg show"
  id="scroll-to-top"
  onClick={[MockFunction]}
>
  <svg
    className="align-text-bottom text-secondary"
    fill="currentColor"
    height={18}
    viewBox="0 0 16 16"
    width={18}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15a.5.5 0 0 0 .5-.5V2.707l3.146 3.147a.5.5 0 0 0 .708-.708l-4-4a.5.5 0 0 0-.708 0l-4 4a.5.5 0 1 0 .708.708L7.5 2.707V14.5a.5.5 0 0 0 .5.5z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
